Helper/MetricsClient.php

```diff
@@ -341,9 +352,19 @@ public function postMetrics()
         if (!$this->configHelper->shouldCaptureMetrics()) {
             return null;
         }
+        $previousPostTime = $this->loadFromCache(self::METRICS_TIMESTAMP_ID);
+        // checks if key exists
+        if (!$previousPostTime) {
+            $this->saveToCache(self::METRICS_TIMESTAMP_ID, round(microtime(true) * 1000));
+        } else {
+            $timeDiff = round(microtime(true) * 1000) - $previousPostTime;
+            // if its under 30 seconds since last key don't do anything
```

Comment:
let's avoid mentioning "30 seconds" - you may update constant one day and this comment will get stale. we can say 'if it's under the threshold' or something.